internal/providers/terraform/aws/waf_web_acl.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/tidwall/gjson"
+)
+
+func GetWafWebACLRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_waf_web_acl",
+		RFunc: NewWafWebACL,
+		Notes: []string{
+			"Seller fees for Managed Rule Groups from AWS Marketplace are not included. Bot Control is not supported by Terraform.",
+		},
+	}
+}
+
+func NewWafWebACL(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	region := d.Get("region").String()
+
+	var costComponents []*schema.CostComponent
+	var ruleGroupRules, monthlyRequests, rule *decimal.Decimal
+	var ruleforGroup int
+
+	costComponents = append(costComponents, wafWebACLCostComponent(
+		region,
+		"Web ACL usage",
+		"months",
+		"USE1-WebACL",
+		1,
+	))
+	var rules []gjson.Result
+	if d.Get("rules").Type != gjson.Null {
+		rules = d.Get("rules").Array()
+		var count int
+		for _, val := range rules {
+			types := val.Get("type").String()
+
+			if types == "REGULAR" || types == "RATE_BASED" {
+				count++
+			}
+
+		}
+		rule = decimalPtr(decimal.NewFromInt(int64(count)))
+	}
+
+	if u != nil && u.Get("rule_group_rules").Type != gjson.Null {
+		ruleGroupRules = decimalPtr(decimal.NewFromInt(u.Get("rule_group_rules").Int()))
+		sum := ruleGroupRules.Add(*rule)
+		costComponents = append(costComponents, wafWebACLUsageCostComponent(
+			region,
+			"Rules",
+			"months",
+			"USE1-Rule",
+			1,
+			&sum,
+		))
+	}
+
+	if ruleGroupRules == nil {
+		costComponents = append(costComponents, wafWebACLUsageCostComponent(
+			region,
+			"Rules",
+			"months",
+			"USE1-Rule",
+			1,
+			rule,
+		))
+	}
+
+	rules = d.Get("rules").Array()
+	var count int
+	for _, val := range rules {
+		types := val.Get("type").String()
+		if types == "GROUP" {
+			count++
+		}
+	}
+	ruleforGroup = count
+	if ruleforGroup > 0 {
+		costComponents = append(costComponents, wafWebACLCostComponent(
+			region,
+			"Rule groups",
+			"months",
+			"USE1-Rule",
+			ruleforGroup,
+		))
+	}
+
+	if u != nil && u.Get("monthly_requests").Type != gjson.Null {
+		monthlyRequests = decimalPtr(decimal.NewFromInt(u.Get("monthly_requests").Int()))
+	}
+
+	costComponents = append(costComponents, wafWebACLUsageCostComponent(
+		region,
+		"Requests",
+		"1M requests",
+		"USE1-Request",
+		1000000,
+		monthlyRequests,
+	))
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func wafWebACLUsageCostComponent(region, displayName, unit, usagetype string, unitMultiplier int, quantity *decimal.Decimal) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:            displayName,
+		Unit:            unit,
+		UnitMultiplier:  unitMultiplier,
+		MonthlyQuantity: quantity,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("aws"),
+			Region:        strPtr(region),
+			Service:       strPtr("awswaf"),
+			ProductFamily: strPtr("Web Application Firewall"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "usagetype", Value: strPtr(usagetype)},
```

Comment:
Now anytime we see `Value` or `ValueRegex` with a variable, we should take care of case-sensitive lookups to follow [contribution guide](https://github.com/infracost/infracost/blob/master/CONTRIBUTING.md#usage-file-guidelines).
```suggestion
				{Key: "usagetype", Value: strPtr(fmt.Sprintf("/%s/i", usagetype))},
```

Reply:
To match the complete value we can use `^` and `$` symbols: ValueRegex: strPtr(fmt.Sprintf("/^%s$/i", usagetype))},